### PR TITLE
a few new actions and some minor bugfix

### PR DESCRIPTION
**Pull request by qudongfang**
## new feature
1. add file/upload
1. add a fully customizable create-scan

## bugfix
1. bugfix of list policy templates
1. bugfix of list scanners
1. bugfix of list scan templates

## improvement
1. add more fields into resources

## Background:
I'm working on:
1. automating the policy updating process
1. scan job creation process

with an internal service(golang) which will listen to the changes of internal scanning policy config repo.
I'm leveraging your work as an SDK.

Nessus API Documentation v. 8.12.1

### CODE REVIEW

File: nessie.go [hunk 1]
```diff
@@ -922,7 +937,36 @@ func (n *nessusImpl) Permissions(objectType string, objectID int64) ([]Permissio
...
+	defer resp.Body.Close()
+	var reply CreatePolicyResp
+	if err = json.NewDecoder(resp.Body).Decode(&reply); err != nil {
+		return CreatePolicyResp{}, err
+	}
+	return reply, nil
+}
+
+// ConfigurePolicy Changes the parameters of a policy.
+func (n *nessusImpl) ConfigurePolicy(policyID int64, createFolderRequest CreatePolicyRequest) error {
```
Comment (attwad) [inline nessie.go]: Can you change `createFolderRequest` into simply `req`? Seems like bad copy&paste

File: nessie.go [hunk 2]
```diff
@@ -932,6 +976,65 @@ func (n *nessusImpl) DeletePolicy(policyID int64) error {
...
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("Filedata", filepath.Base(filePath))
+	if err != nil {
+		return err
+	}
+	_, err = io.Copy(part, f)
+
+	err = writer.Close()
```
Comment (attwad) [inline nessie.go]: prefer one line when possible:
```
if err := writer.Close(); err != nil {
  return err
}
```

**Review by attwad — APPROVED** [review]

File: nessie.go (hunk 1 above)
Comment (qudongfang) [reply nessie.go]: Nice catch, Certainly!

File: nessie.go (hunk 2 above)
Comment (qudongfang) [reply nessie.go]: Yeah, it's cleaner.

**Comment by qudongfang** [comment]
Got another commit. 
Thanks for the review!

**Review by attwad — APPROVED** [review]

**Comment by attwad** [comment]
Cheers!

**Pull request merged**